### ISSUE TITLE
Cant run two builds in the same process

### ISSUE THREAD

**Issue by Klaven** [buildpacks/pack#1917]
### Summary

I have been trying to build two different images and using github.com/buildpacks/pack as a lib. That being said I am unable to run two builds, I have to stop the process and start it again to build the second container.

When the second build starts this error is almost immediately returned:
`executing lifecycle: write /dev/stderr: file already closed`
This seems to revolve around the use of os.Stderr.

---

### Reproduction
I'm just trying to build two consecutive containers, here is a simple demo.
```go
func main() {

	ctx := context.Background()

	c, err := client.NewClient()
	if err != nil {
		logrus.Error(err)
	}

	envs := make(map[string]string)

	envs["CGO_ENABLED"] = "0"
	envs["BP_GO_BUILD_FLAGS"] = "-buildmode=default"

	// should be the equivalent of  pack build my-app  --buildpack paketo-buildpacks/go  --builder paketobuildpacks/builder-jammy-buildpackless-static
	buildOpts := client.BuildOptions{
		GroupID:    -1,
		Image:      "web-example",
		Builder:    "paketobuildpacks/builder-jammy-full",
		Buildpacks: []string{"paketo-buildpacks/web-servers"},
		PullPolicy: image.PullIfNotPresent,
		AppPath:    "./sample/public/",
		// DockerHost: "unix:///var/run/docker.sock",
		Env: map[string]string{
			"BP_WEB_SERVER": "nginx",
		},
	}

	err = c.Build(ctx, buildOpts)
	if err != nil {
		logrus.Error(err)
	}

	cli, err := client.NewClient()

	buildOpts = client.BuildOptions{
		GroupID:    -1,
		Image:      "web-example-2",
		Builder:    "paketobuildpacks/builder-jammy-full",
		Buildpacks: []string{"paketo-buildpacks/web-servers"},
		PullPolicy: image.PullIfNotPresent,
		AppPath:    "./sample/public/",
		// DockerHost: "unix:///var/run/docker.sock",
		Env: map[string]string{
			"BP_WEB_SERVER": "nginx",
		},
	}

	// Will error here
	err = cli.Build(ctx, buildOpts)
	if err != nil {
		logrus.Error(err)
		fmt.Println(err.Error())
	}

}
```

##### Steps

1. Build two containers one after the other

##### Current behavior

I get the following error: executing lifecycle: write /dev/stderr: file already closed

##### Expected behavior

To build a second container

---

##### pack info

Pack:
  Version:  0.0.0
  OS/Arch:  linux/amd64

Default Lifecycle Version:  0.11.3

Supported Platform APIs:  0.3, 0.4, 0.5, 0.6

##### docker info

Client:
 Version:    24.0.5
 Context:    default
 Debug Mode: false

Server:
 Containers: 46
  Running: 2
  Paused: 0
  Stopped: 44
 Images: 224
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 091922f03c2762540fd057fba91260237ff86acb.m
 runc version:
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.5.3-arch1-1
 Operating System: Arch Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 32
 Total Memory: 61.94GiB
 Name: archlinux
 ID: GISC:YUBA:ZWBG:BP3N:5J2T:F63C:5SCO:FLQR:ZPE4:SAVM:62XJ:RQ6W
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: countsm
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by natalieparellano** [comment]
I wonder if this is related (somehow) to https://github.com/buildpacks/pack/issues/1214 - although that mentions stdout.

**Comment by Klaven** [comment]
Yes, I have messed around with it a bit and #1214 does in fact seem to be talking about the same thing. I have investigated a bit further and it is the stdout that is being used by the logging that is getting closed causing the second build to fail. We can close this for the original issue as I do believe that it is still an issue, though I did not look into the logic behind it it seems to be the same issue. Thank you!
